packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx

```diff
@@ -148,19 +148,48 @@ export const ChatMessageComponentAsRichTextEditBox = (
     // get plain text content from the editor to check if the message is empty
     // as the content may contain tags even when the content is empty
     const plainTextContent = editTextFieldRef.current?.getPlainContent() ?? contentValue;
+    /* @conditional-compile-remove(rich-text-editor-image-upload) */
+    const imageIds = inlineImageIds(contentValue);
+    /* @conditional-compile-remove(rich-text-editor-image-upload) */
+    if (imageIds.length > 0) {
+      // Check for changes
+      const existingIds = inlineImageIds(message.content);
+      // For mismatching array sizes, something has changed with images. OK to submit.
+      // Note that the case of zero images falls back to the character length check
+      if (existingIds.length !== imageIds.length) {
```

Comment:
Do we need an additional array processing here? as long as there are images, content won't be in state "too short"

Also, 
```
      if (!arraysMatch) {
        return 'OK';
      }
```
might return "Ok" status when content is too long